src/components/TodoItem.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+interface TodoItemProps {
+  filteredTodos: Todo[];
+  tempClearAll: Todo[] | null;
+  tempDeleteTodo: Todo | null;
+  handleDelete: (id: number) => void;
+}
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  filteredTodos,
+  tempClearAll,
+  tempDeleteTodo,
+  handleDelete,
+}) => {
+  return (
+    <>
+      {filteredTodos.map(todo => {
+        return (
+          <>
+            {/* overlay will cover the todo while it is being deleted or updated */}
+
+            {/* This todo is an active todo */}
+            <div
+              data-cy="Todo"
+              className={`todo ${todo.completed && 'completed'}`}
+              key={todo.id}
+            >
+              <label className="todo__status-label" htmlFor={`todo-${todo.id}`}>
+                {
+                  <input
+                    data-cy="TodoStatus"
+                    type="checkbox"
+                    className="todo__status"
+                    checked={todo.completed}
+                  />
+                }
+              </label>
+
+              <span data-cy="TodoTitle" className="todo__title">
+                {todo.title}
+              </span>
+              <button
+                type="button"
+                className="todo__remove"
+                data-cy="TodoDelete"
+                onClick={() => {
+                  handleDelete(todo.id);
+                }}
+              >
+                ×
+              </button>
+
+              <div
+                data-cy="TodoLoader"
+                className={`modal overlay ${tempDeleteTodo?.id === todo.id || tempClearAll?.some(t => t.id === todo.id) ? 'is-active' : ''}`}
```

Comment:
You have a few separate variables to define if todo is in loading state. Instead, create one variable for it with an  array of ids (similar to tempClearAll but rename to something more descriptive, like `loadingTodoIds`, no need to store entire todos, ids is enough)
Use this array for all actions (updating todo, deleting todo)

Also, it would be nice to reuse the TodoItem component for tempTodo displaying. So it should accept just `isLoading` prop (will be always true for tempTodo)